metis-media-service/src/main/java/eu/europeana/metis/mediaservice/WebResource.java

```diff
@@ -0,0 +1,160 @@
+import eu.europeana.corelib.definitions.jibx.Type1;
+import eu.europeana.corelib.definitions.jibx.WebResourceType;
+import eu.europeana.corelib.definitions.jibx.Width;
+
+public class WebResource {
+	private final WebResourceType resource;
+	
+	WebResource(WebResourceType resource) {
+		this.resource = resource;
+	}
+	
+	public void setWidth(int width) {
+		resource.setWidth(intVal(new Width(), width));
+	}
+	
+	public void setHeight(int height) {
+		resource.setHeight(intVal(new Height(), height));
+	}
+
+	public void setMimeType(String mimeType) {
+		HasMimeType hasMimeType = new HasMimeType();
+		hasMimeType.setHasMimeType(mimeType);
+		resource.setHasMimeType(hasMimeType);
+	}
+	
+	public void setFileSize(long fileSize) {
+		resource.setFileByteSize(longVal(fileSize));
+	}
+	
+	public void setColorspace(String colorspace) {
+		if (!"grayscale".equals(colorspace) && !"sRGB".equals(colorspace))
+			throw new IllegalArgumentException("Unrecognized color space: " + colorspace);
+		HasColorSpace hasColorSpace = new HasColorSpace();
+		hasColorSpace.setHasColorSpace("sRGB".equals(colorspace) ? ColorSpaceType.S_RGB : ColorSpaceType.GRAYSCALE);
+		resource.setHasColorSpace(hasColorSpace);
+	}
+	
+	public void setOrientation(boolean landscape) {
+		resource.setOrientation(stringVal(new OrientationType(), landscape ? "landscape" : "portrait"));
+	}
+	
+	public void setDominantColors(List<String> dominantColors) {
+		resource.setComponentColorList(dominantColors.stream()
+				.peek(c -> {
+					if (!c.matches("[0-9A-F]{6}"))
+						throw new IllegalArgumentException();
+				})
+				.map(c -> "#" + c) // TODO dominant colors start with '#' due to legacy systems
+				.map(c -> {
+					HexBinaryType hex = new HexBinaryType();
+					hex.setString(c);
+					hex.setDatatype("http://www.w3.org/2001/XMLSchema#hexBinary");
+					return hex;
+				})
+				.collect(Collectors.toList()));
+	}
+	
+	/** @param duration duration in seconds */
+	public void setDuration(double duration) {
+		Duration duration2 = new Duration();
+		duration2.setDuration(Integer.toString((int) Math.round(duration * 1000)));
+		resource.setDuration(duration2);
+	}
+	
+	public void setBitrate(int bitrate) {
+		resource.setBitRate(uintVal(new BitRate(), bitrate));
+	}
+
+	public void setFrameRete(double frameRate) {
+		resource.setFrameRate(doubleVal(frameRate));
+	}
+
+	public void setCodecName(String codecName) {
+		CodecName codecName2 = new CodecName();
+		codecName2.setCodecName(codecName);
+		resource.setCodecName(codecName2);
+	}
+	
+	public void setCahhnels(int channels) {
+		resource.setAudioChannelNumber(uintVal(new AudioChannelNumber(), channels));
+	}
+	
+	public void setSampleRate(int sampleRate) {
+		resource.setSampleRate(intVal(new SampleRate(), sampleRate));
+	}
+	
+	public void setSampleSize(int sampleSize) {
+		resource.setSampleSize(intVal(new SampleSize(), sampleSize));
+	}
+	
+	public void setContainsText(boolean containsText) {
```

Comment:
Set here where two methods would be cleaner than using one method with a boolean parameter.

Reply:
This rule doesn't make sense here. `setContainsText(true)` and `setContainsText(false)` are obvious to read. Furthermore, where I call it I already have a boolean variable and I don't want to introduce unnecessary if-else.